Owl/app/src/main/java/com/example/owl/ui/courses/Courses.kt

```diff
@@ -21,66 +21,50 @@ import androidx.annotation.StringRes
-import androidx.compose.material.MaterialTheme
-import androidx.compose.material.Scaffold
-import androidx.compose.material.Text
 import androidx.compose.material.TopAppBar
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.filled.AccountCircle
-import androidx.compose.material.primarySurface
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.State
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.navigate
 import com.example.owl.R
 import com.example.owl.model.courses
 import com.example.owl.model.topics
-import com.example.owl.ui.theme.BlueTheme
-import dev.chrisbanes.accompanist.insets.navigationBarsHeight
-import dev.chrisbanes.accompanist.insets.navigationBarsPadding
+import com.example.owl.ui.MainDestinations
 
-@Composable
-fun Courses(selectCourse: (Long) -> Unit) {
-    BlueTheme {
-        val (selectedTab, setSelectedTab) = remember { mutableStateOf(CourseTabs.FEATURED) }
-        val tabs = CourseTabs.values()
-        Scaffold(
-            backgroundColor = MaterialTheme.colors.primarySurface,
-            bottomBar = {
-                BottomNavigation(
-                    Modifier.navigationBarsHeight(additional = 56.dp)
-                ) {
-                    tabs.forEach { tab ->
-                        BottomNavigationItem(
-                            icon = { Icon(painterResource(tab.icon), contentDescription = null) },
-                            label = { Text(stringResource(tab.title).toUpperCase()) },
-                            selected = tab == selectedTab,
-                            onClick = { setSelectedTab(tab) },
-                            alwaysShowLabel = false,
-                            selectedContentColor = MaterialTheme.colors.secondary,
-                            unselectedContentColor = LocalContentColor.current,
-                            modifier = Modifier.navigationBarsPadding()
-                        )
-                    }
-                }
-            }
-        ) { innerPadding ->
-            val modifier = Modifier.padding(innerPadding)
-            when (selectedTab) {
-                CourseTabs.MY_COURSES -> MyCourses(courses, selectCourse, modifier)
-                CourseTabs.FEATURED -> FeaturedCourses(courses, selectCourse, modifier)
-                CourseTabs.SEARCH -> SearchCourses(topics, modifier)
+fun NavGraphBuilder.courses(
+    onCourseSelected: (Long) -> Unit,
+    onboardingComplete: State<Boolean>,
```

Comment:
Why does this need to be a `State`? Seems like if could just be a `Boolean`?

Reply:
Yeah this was the main blocker for this PR. I _think_ that the problem is that this is not a Composable so passing a boolean is not going to get recomposition magic as we expect.

Reply:
Added https://issuetracker.google.com/174783110 to the code